weave/tests/test_index_sqlite.py

```diff
@@ -0,0 +1,204 @@
+"""Pytest tests for the index directory."""
+import os
+from datetime import datetime
+
+import pytest
+import s3fs
+from fsspec.implementations.local import LocalFileSystem
+
+from weave.pantry import Pantry
+from weave.index.index_sqlite import IndexSQLite
+from weave.tests.pytest_resources import PantryForTest
+
+
+###############################################################################
+#                      Pytest Fixtures Documentation:                         #
+#            https://docs.pytest.org/en/7.3.x/how-to/fixtures.html            #
+#                                                                             #
+#                  https://docs.pytest.org/en/7.3.x/how-to/                   #
+#          fixtures.html#teardown-cleanup-aka-fixture-finalization            #
+#                                                                             #
+#  https://docs.pytest.org/en/7.3.x/how-to/fixtures.html#fixture-parametrize  #
+###############################################################################
+
+# Pylint doesn't like redefining the test fixture here from
+# test_basket, but this is the right way to do it if at some
+# point in the future the two need to be differentiated.
+# pylint: disable=duplicate-code
+
+s3fs = s3fs.S3FileSystem(
+    client_kwargs={"endpoint_url": os.environ["S3_ENDPOINT"]}
+)
+local_fs = LocalFileSystem()
+
+
+# Test with two different fsspec file systems (above).
+@pytest.fixture(
+    name="test_pantry",
+    params=[s3fs, local_fs],
+    ids=["S3FileSystem", "LocalFileSystem"],
+)
+def fixture_test_pantry(request, tmpdir):
+    """Sets up test pantry for the tests"""
+    file_system = request.param
+    test_pantry = PantryForTest(tmpdir, file_system)
+    yield test_pantry
+    test_pantry.cleanup_pantry()
+
+
+def test_index_is_different_between_pantries(test_pantry):
+    """Validate that pantries will have their own index/db file instead of
+    sharing one
+    """
+    pantry_1_path = os.path.join(test_pantry.pantry_path, "test-pantry-1")
+    pantry_2_path = os.path.join(test_pantry.pantry_path, "test-pantry-2")
+
+    test_pantry.file_system.mkdir(pantry_1_path)
+    test_pantry.file_system.mkdir(pantry_2_path)
+
+    tmp_txt_file = test_pantry.tmpdir.join("test.txt")
+    tmp_txt_file.write("this is a test")
+
+    upload_path_1 = os.path.join(pantry_1_path, "text.txt")
+    upload_path_2 = os.path.join(pantry_2_path, "text.txt")
+
+    test_pantry.file_system.upload(str(tmp_txt_file.realpath()), upload_path_1)
+    test_pantry.file_system.upload(str(tmp_txt_file.realpath()), upload_path_2)
+
+    # Make the pantries.
+    pantry_1 = Pantry(
+        IndexSQLite,
+        pantry_path=pantry_1_path,
+        file_system=test_pantry.file_system,
+        sync=True,
+    )
+
+    pantry_2 = Pantry(
+        IndexSQLite,
+        pantry_path=pantry_2_path,
+        file_system=test_pantry.file_system,
+        sync=True,
```

Comment:
I believe sync=True is only used in IndexPandas. Not required for sqlite, remove here and elsewhere.

Reply:
I believe I removed them all